tests/unit/test_main_cli.py

```diff
@@ -44,6 +44,70 @@ def test_github_auth_env() -> None:
     assert result.exit_code == 0
 
 
+@patch("dd_license_attribution.cli.main_cli.GitHub")
+@patch("dd_license_attribution.cli.main_cli.SourceCodeManager")
+@patch("dd_license_attribution.cli.main_cli.PythonEnvManager")
+@patch("dd_license_attribution.cli.main_cli.MetadataCollector")
+def test_skip_strategies_options(
+    mock_metadata_collector: Mock,
+    mock_python_env_manager: Mock,
+    mock_source_code_manager: Mock,
+    mock_github: Mock,
+) -> None:
+
+    mock_metadata_collector.return_value.collect_metadata.return_value = []
+
+    test_cases = [
+        (
+            ["--no-pypi-strategy"],
+            "PythonPipMetadataCollectionStrategy",
+        ),
+        (
+            ["--no-gopkg-strategy"],
+            "GoPkgsMetadataCollectionStrategy",
+        ),
+        (
+            ["--no-github-sbom-strategy"],
+            "GitHubSbomMetadataCollectionStrategy",
+        ),
+    ]
```

Comment:
We could fit this in a `@pytest.mark.parametrize` as we do in test_source_code_manager to be more explicit about intention.